File: app/models/equipment.rb
class Equipment < ActiveRecord::Base
  serialize :rhand
  serialize :lhand
  serialize :chest
  serialize :pants
  serialize :head 

 belongs_to :character

  def items
    slots={}
    self.attributes.each do |slot,val|
      if slot !~ /^.*id|at$/ && val
        slots[slot]=val
      end
    end
    slots
  end
  def stats
    stat={}
    self.attributes.each do |slot,val|
      if slot !~ /^.*id|at$/ && val
        val[:stats].each do |s,v|
          stat[s.to_sym]?stat[s.to_sym]+=v:stat[s.to_sym]=v
        end
      end
    end
    stat
  end
end

File: public/javascripts/tufflife/controllers/equipment.js
dojo.provide("tufflife.controllers.equipment");
dojo.require("tufflife.controllers.utils.itemUtils")


dojo.declare("equipment", null, {
   setItems: function(){
       var slots = ["rhand","lhand","head","chest","pants"];
       dojo.forEach(slots, function(slot){
          var tbCell = dojo.byId(slot);
          dojo.empty(tbCell)
          dojo.create("img",{src:"images/emptyslot.png",id:"eq_"+slot},tbCell,"first")
       });
       tufflife.data.itemstore.fetch({
           query: {location:"equipment"},
           onComplete: function(items){
               dojo.forEach(items, function(item){
                   var slot = tufflife.data.itemstore.getValue(item,"slot");
                   var id = "eq_"+slot
                   var image = tufflife.data.itemstore.getValue(item,"image");
                   var tbCell = dojo.byId(slot)
                   dojo.empty(tbCell);
                   dojo.create("img",{src:image,id:id},tbCell,"first");

                   setItemCtls(item,id);
               })
           }
       })
   },
   constructor: function(){
       this.setItems();
   }
});
tufflife.controllers.equipment = new equipment();